src/test/java/org/springframework/data/jpa/repository/UserRepositoryStoredProcedureTests.java

```diff
@@ -74,6 +86,33 @@ public void entityAnnotatedCustomNamedProcedurePlus1IO() {
 		assertThat(repository.entityAnnotatedCustomNamedProcedurePlus1IO(1), is(2));
 	}
 
+	@Test // DATAJPA-707
+	public void entityAnnotatedCustomNamedProcedurePlus1IOInvalidOutParamName() {
+
+		assumeTrue(currentEntityManagerIsAJpa21EntityManager(em));
+
+		exception.expect(InvalidDataAccessApiUsageException.class);
+		exception.expectMessage(containsString("Could not locate parameter registered under that name"));
+
+		repository.entityAnnotatedCustomNamedProcedurePlus1IOInvalidOutParamName(1);
+	}
+
+	@Test // DATAJPA-707
```

Comment:
Added this test to verify that the pre-existing functionality still works, as it appears no test previously hit this case.